src/main/scala/reporting/Converter.scala

```diff
@@ -0,0 +1,511 @@
+package viper.silicon.reporting
+
+import scala.util.{Try, Success}
+import viper.silver.verifier.{
+  Model,
+  ModelEntry,
+  ValueEntry,
+  ConstantEntry,
+  ApplicationEntry,
+  MapEntry
+}
+import viper.silver.ast
+import viper.silicon.interfaces.state.Chunk
+import viper.silicon.resources.{FieldID, PredicateID}
+import viper.silicon.state.{Store, State, BasicChunk}
+import viper.silicon.state.terms.{
+  sorts,
+  Sort,
+  Term,
+  Unit,
+  IntLiteral,
+  BooleanLiteral,
+  Null,
+  Var,
+  App,
+  Combine,
+  First,
+  Second,
+  SortWrapper,
+  PredicateLookup,
+  Rational,
+  PermLiteral
+}
+/* Explanation: (to be removed later)
+    To use these new extracted counterexamples, you can use the flag "--counterexample mapped"
+    The new trait ExtractedModelEntry should make the counterexamples more readable and extract
+    information from the heap, but heap values such as predicates can still be accessed,
+    also in a more processed way since all the first/second/Sortwrappers should be gone now and
+    replaced with their corresponding values in the extracted Heap fields of the converter class
+
+    A lot of types like sequences etc are not handled yet, and I am not sure if I plan to do so
+    since they are not used in Rust
+
+    The various parts in this file should probably be moved out of the interfaces package,
+    but for now to keep everything in one place is simpler to work on it.
+
+    To see the advantage of these new counterexamples you can run it on the
+    files in src/test/resourcers/demo-counterexamples, the file simple-refs-rec.vpr is
+    probably the most interesting.
+
+ */
```

Comment:
To be removed? I guess this is more of a PR description.

Reply:
Oh yes, sure. I also just removed all println's.